integration-tests/integration_test.go

```diff
@@ -468,6 +480,85 @@ func TestChildPrefix(t *testing.T) {
 	helper.Logf("Pinging %s from node2", node3LoopbackIP)
 	err = ping(ctx, node2, inetV4, node3LoopbackIP.String())
 	require.NoError(err)
+
+	helper.Logf("Pinging %s from node3", node2LoopbackIPv6)
+	err = ping(ctx, node3, inetV6, node2LoopbackIPv6.String())
+	require.NoError(err)
+
+	helper.Logf("Pinging %s from node2", node3LoopbackIPv6)
+	err = ping(ctx, node2, inetV6, node3LoopbackIPv6.String())
+	require.NoError(err)
+
+	// kill the nexodus process on both nodes
+	_, err = helper.containerExec(ctx, node1, []string{"killall", "nexd"})
+	require.NoError(err)
+	_, err = helper.containerExec(ctx, node2, []string{"killall", "nexd"})
+	require.NoError(err)
+	_, err = helper.containerExec(ctx, node3, []string{"killall", "nexd"})
+	require.NoError(err)
+
+	// start nexd two more times, only validate connectivity on the second.
+	for i := 0; i < 2; i++ {
+		// start nexodus on the nodes
+		helper.runNexd(ctx, node1,
+			"--username", username, "--password", password,
+			"relay",
+		)
+
+		// validate nexd has started on the relay node
+		err := helper.nexdStatus(ctx, node1)
+		require.NoError(err)
+
+		helper.runNexd(ctx, node2,
+			"--username", username, "--password", password,
+			"router", fmt.Sprintf("--child-prefix=%s", node2ChildPrefix),
+		)
+
+		helper.runNexd(ctx, node3,
+			"--username", username, "--password", password,
+			"router", fmt.Sprintf("--child-prefix=%s", node3ChildPrefix),
+		)
+
+		// address will be the same, this is just a readiness check for gather data
+		node3IP, err = getContainerIfaceIP(ctx, inetV4, "wg0", node3)
+		require.NoError(err)
+		node2IP, err = getContainerIfaceIP(ctx, inetV4, "wg0", node2)
+		require.NoError(err)
```

Comment:
If you don't specifically care about getting it from `wg0`, there is also a `getTunnelIP()` helper that will get you the same IP, but via `nexctl nexd get tunnelip`. Using that interface should work for all configurations (normal mode, proxy mode, future drivers that aren't wg0 ...).

A major difference is that `getTunnelIP()` does not do retries, while right now `getContainerIfaceIP()` does. We've got retries in multiple places all dealing with the same issue (`nexd` might not be up yet). I think it would nice to simplify by having one place that does that, probably `nexdStatus()`.